zkevm-circuits/src/evm_circuit/execution/begin_tx.rs

```diff
@@ -102,6 +113,18 @@ impl<F: Field> ExecutionGadget<F> for BeginTxGadget<F> {
             None,
         );
 
+        // is_nonce_valid = instruction.is_zero(tx_nonce - nonce_prev)
+        let is_nonce_valid = IsZeroGadget::construct(cb, tx_nonce.expr() - tx_nonce.expr());
+
+        // instruction.constrain_equal(nonce, nonce_prev.expr() + 1 - is_tx_invalid)
+        // nonce_prev = tx_nonce.expr()
+        // bump the account nonce if the tx is valid
+        cb.require_equal(
+            "nonce, nonce_prev and invalid_tx_tx",
+            tx_nonce.expr() + 1.expr(),
+            tx_nonce.expr() + 1.expr() - is_tx_invalid.expr(),
+        );
```

Comment:
Change this to follow the spec when you have the correct nonce values from above.